### PR TITLE
add object id (ar-2805) and refactor message format (ar-2756)

### PR DESCRIPTION
**Pull request by jae1e**

### CODE REVIEW

File: msg/LineRoadBoundary.msg [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
+uint8 position
+uint8 type
 float64 position_parameter_c0
 float64 heading_angle_parameter_c1
 float64 curvature_parameter_c2  
 float64 curvature_derivative_parameter_c3
-uint16 age # please specify time unit
-float32 cofidence_measure
+uint16 age
+float32 cofidence
```
Comment (sebastian-seoul-robotics) [inline msg/LineRoadBoundary.msg]: confidence

File: msg/ObjectProperties.msg [hunk 2]
```diff
@@ -1,4 +1,16 @@
...
+
+# Status enum
+uint8 STATUS_DYNAMIC=0
+uint8 STATUS_STATIC=1
+
+int32 id
+uint8 type
+uint8 status
+uint16 age
+float32 cofidence
```
Comment (sebastian-seoul-robotics) [inline msg/ObjectProperties.msg]: confidence

File: msg/Object.msg [hunk 3]
```diff
@@ -0,0 +1,4 @@
+std_msgs/Header header # header.timestamp: same as the input point cloud message this message was generated from
```
Comment (sebastian-seoul-robotics) [inline msg/Object.msg]: Do we need this header here? Is it different from `ObjectList/header`?

File: msg/LaneList.msg [hunk 4]
```diff
@@ -1,7 +1,4 @@
-std_msgs/Header header
-  uint32 seq
-  time stamp # same as the input point cloud message this message was generated from
-  string frame_id
-time publication_stamp # the time when this message was published
+std_msgs/Header header # header.timestamp: same as the input point cloud message this message was generated from
+time publication_stamp # publication_stamp: the time when this message was published
```
Comment (sebastian-seoul-robotics) [inline msg/LaneList.msg]: time publication_stamp # the time when this message was published

File: msg/LaneList.msg [hunk 5]
```diff
@@ -1,7 +1,4 @@
...
 sensr_msgs/LineRoadBoundary[] lines
```
Comment (sebastian-seoul-robotics) [inline msg/LaneList.msg]: `LineRoadBoundary` sounds very weird to me. Maybe `LaneLine` or `RoadLine`?

File: msg/LaneList.msg [hunk 6]
```diff
@@ -1,7 +1,4 @@
...
 sensr_msgs/LineRoadBoundary[] road_boundaries
```
Comment (sebastian-seoul-robotics) [inline msg/LaneList.msg]: curbside

File: msg/LineRoadBoundary.msg [hunk 7]
```diff
@@ -1,13 +1,22 @@
...
+uint8 POSITION_NEXT_RIGHT=3
+
+# Type enum
+uint8 TYPE_LINE=0
+uint8 TYPE_ROAD_BOUNDARY_POSITIVE_ELEVATION=1
+uint8 TYPE_ROAD_BOUNDARY_NEGATIVE_ELEVATION=2
+uint8 TYPE_ROAD_BOUNDARY_FLAT=3
+
+std_msgs/Header header # header.timestamp: same as the input point cloud message this message was generated from
+time publication_stamp # publication_stamp: the time when this message was published
```
Comment (sebastian-seoul-robotics) [inline msg/LineRoadBoundary.msg]: time publication_stamp # the time when this message was published

File: msg/ObjectList.msg [hunk 8]
```diff
@@ -1,7 +1,3 @@
-std_msgs/Header header
-  uint32 seq
-  time stamp # same as the input point cloud message this message was generated from
-  string frame_id
-time publication_stamp # the time when this message was published
-sensr_msgs/ObjectKinematics[] object_kinematics
-sensr_msgs/ObjectProperties[] object_properties
+std_msgs/Header header # header.timestamp: same as the input point cloud message this message was generated from
+time publication_stamp # publication_stamp: the time when this message was published
```
Comment (sebastian-seoul-robotics) [inline msg/ObjectList.msg]: time publication_stamp # the time when this message was published

File: msg/Object.msg [hunk 9]
```diff
@@ -0,0 +1,4 @@
...
+time publication_stamp # publication_stamp: the time when this message was published
```
Comment (sebastian-seoul-robotics) [inline msg/Object.msg]: time publication_stamp # the time when this message was published

File: msg/LineRoadBoundary.msg [hunk 10]
```diff
@@ -1,13 +1,22 @@
...
+uint8 position
+uint8 type
```
Comment (sebastian-seoul-robotics) [inline msg/LineRoadBoundary.msg]: road_boundary_type

File: msg/LineRoadBoundary.msg [hunk 11]
```diff
@@ -1,13 +1,22 @@
...
+uint8 POSITION_NEXT_LEFT=2
+uint8 POSITION_NEXT_RIGHT=3
+
+# Type enum
+uint8 TYPE_LINE=0
+uint8 TYPE_ROAD_BOUNDARY_POSITIVE_ELEVATION=1
+uint8 TYPE_ROAD_BOUNDARY_NEGATIVE_ELEVATION=2
+uint8 TYPE_ROAD_BOUNDARY_FLAT=3
+
+std_msgs/Header header # header.timestamp: same as the input point cloud message this message was generated from
```
Comment (sebastian-seoul-robotics) [inline msg/LineRoadBoundary.msg]: std_msgs/Header header # header.timestamp: timestamp of the input cloud message

File: msg/ObjectKinematics.msg [hunk 12]
```diff
@@ -1,8 +1,3 @@
-std_msgs/Header header
-  uint32 seq
-  time stamp # same as the input point cloud message this message was generated from
-  string frame_id
-time publication_stamp # the time when this message was published
 geometry_msgs/Pose pose
 geometry_msgs/Twist twist
```
Comment (sebastian-seoul-robotics) [inline msg/ObjectKinematics.msg]: I remember that we had a discussion that twist is not very known. Maybe `geometry_msgs/Twist velocity` might be more straightforward for the user

File: msg/ObjectProperties.msg [hunk 13]
```diff
@@ -1,4 +1,16 @@
-uint8 obstacle_type # 0: 4 Wheeler large 1: 4 Wheeler small 2: 2 Wheeler 3: Pedestrian 4: Unknown
```
Comment (sebastian-seoul-robotics) [inline msg/ObjectProperties.msg:1]: I personally prefer these more detailed variable names

File: msg/LineRoadBoundary.msg [hunk 14]
```diff
@@ -1,13 +1,22 @@
...
+time publication_stamp # publication_stamp: the time when this message was published
+uint8 position
```
Comment (sebastian-seoul-robotics) [inline msg/LineRoadBoundary.msg]: maybe `relative_position`?

File: msg/LaneList.msg [hunk 15]
```diff
@@ -1,7 +1,4 @@
-std_msgs/Header header
-  uint32 seq
-  time stamp # same as the input point cloud message this message was generated from
-  string frame_id
-time publication_stamp # the time when this message was published
+std_msgs/Header header # header.timestamp: same as the input point cloud message this message was generated from
```
Comment (sebastian-seoul-robotics) [inline msg/LaneList.msg]: std_msgs/Header header # header.timestamp: timestamp of the input cloud message

File: msg/LineRoadBoundary.msg [hunk 16]
```diff
@@ -1,13 +1,22 @@
...
-  uint32 seq
-  time stamp # same as the input point cloud message this message was generated from
-  string frame_id
-time publication_stamp # the time when this message was published
-uint8 position # 0: Ego left 1: Ego right 2: Next left 3: Next right
-uint8 type # 0: Line 1: Road boundary Positive elevation 2: Road boundary Negative elevation 3: Road boundary flat
+# Position enum
+uint8 POSITION_EGO_LEFT=0
+uint8 POSITION_EGO_RIGHT=1
+uint8 POSITION_NEXT_LEFT=2
```
Comment (sebastian-seoul-robotics) [inline msg/LineRoadBoundary.msg]: What does `POSITION_NEXT_LEFT` and `POSITION_NEXT_RIGHT` mean?

File: msg/ObjectProperties.msg [hunk 17]
```diff
@@ -1,4 +1,16 @@
...
+uint8 TYPE_CYCLIST=2
+uint8 TYPE_PEDESTRIAN=3
+uint8 TYPE_UNKNOWN=4
+
+# Status enum
+uint8 STATUS_DYNAMIC=0
+uint8 STATUS_STATIC=1
+
+int32 id
+uint8 type
```
Comment (sebastian-seoul-robotics) [inline msg/ObjectProperties.msg]: The `confidence` refers to the `type`. I think we should group these two. Currently this is not obvious

File: msg/ObjectList.msg [hunk 18]
```diff
@@ -1,7 +1,3 @@
-std_msgs/Header header
-  uint32 seq
-  time stamp # same as the input point cloud message this message was generated from
-  string frame_id
-time publication_stamp # the time when this message was published
-sensr_msgs/ObjectKinematics[] object_kinematics
-sensr_msgs/ObjectProperties[] object_properties
+std_msgs/Header header # header.timestamp: same as the input point cloud message this message was generated from
```
Comment (sebastian-seoul-robotics) [inline msg/ObjectList.msg]: std_msgs/Header header # header.timestamp: timestamp of the input cloud message

File: msg/LineRoadBoundary.msg [hunk 19]
```diff
@@ -1,13 +1,22 @@
...
+uint8 POSITION_NEXT_RIGHT=3
+
+# Type enum
+uint8 TYPE_LINE=0
```
Comment (sebastian-seoul-robotics) [inline msg/LineRoadBoundary.msg]: uint8 TYPE_ROAD_LINE=0

**Review by sebastian-seoul-robotics — COMMENTED** [review]
I left some more comments to clean this repo

**Review by DK-sr — COMMENTED** [review]
It looks good to me :+1: I think it is good to go once the suggestions from Seb are done.

File: msg/LaneList.msg (hunk 5 above)
Comment (jae1e) [reply msg/LaneList.msg]: It was AVL suggestion. I think I'm gonna go with RoadLine and RoadLineList.

File: msg/LineRoadBoundary.msg (hunk 16 above)
Comment (jae1e) [reply msg/LineRoadBoundary.msg]: Not ego line, but a line next to the ego line. When there are 4 lines, ego lines are 2 inner lines and the rest two are next lines.

File: msg/LineRoadBoundary.msg (hunk 19 above)
Comment (jae1e) [reply msg/LineRoadBoundary.msg]: Will go with TYPE_LANE_LINE

File: msg/LineRoadBoundary.msg (hunk 16 above)
Comment (sebastian-seoul-robotics) [reply msg/LineRoadBoundary.msg]: Maybe `POSITION_ADJACENT_LEFT` and `POSITION_ADJACENT_RIGHT` then?

Comment (jae1e) [reply msg/LineRoadBoundary.msg]: Sounds better

**Review by sebastian-seoul-robotics — APPROVED** [review]

File: msg/ObjectProperties.msg [hunk 20]
```diff
@@ -1,4 +1,16 @@
...
+# Status enum
+uint8 STATUS_DYNAMIC=0
+uint8 STATUS_STATIC=1
+
+# Type enum
+uint8 TYPE_CAR_LARGE=0
+uint8 TYPE_CAR_SMALL=1
+uint8 TYPE_CYCLIST=2
+uint8 TYPE_PEDESTRIAN=3
+uint8 TYPE_UNKNOWN=4
```
Comment (DK-sr) [inline msg/ObjectProperties.msg:10]: good to see enum :+1:

**Review by DK-sr — APPROVED** [review]
:+1:

**Pull request merged**